torch/csrc/distributed/autograd/init.cpp

```diff
@@ -30,11 +30,26 @@ PyObject* dist_autograd_init(PyObject* /* unused */) {
               "_context_id",
               &DistAutogradContext::context_id,
               py::call_guard<py::gil_scoped_release>())
+          .def(
+              "_recv_functions",
+              [](const DistAutogradContext& ctx) {
+                std::map<int64_t, py::object> funcs;
+                for (const auto& map_entry : ctx.recvFunctions()) {
+                  funcs.emplace(
+                      map_entry.first,
+                      py::reinterpret_steal<py::object>(
+                          torch::autograd::functionToPyObject(
+                              map_entry.second)));
+                }
+                return funcs;
+              })
```

Comment:
Any reason for not releasing the GIL here? If it is for atomic access to `recvFunctions()`, I feel the ctx should have it's own mutex, no?

Reply:
I think we need the GIL for functionToPyObject.